app/src/main/java/com/duckduckgo/app/onboardingdesignexperiment/OnboardingDesignExperimentManager.kt

```diff
@@ -16,20 +16,244 @@
+    }
+
+    override suspend fun fireVisitSitePromptDisplayedNewTabPixel() {
+        onboardingExperimentMetricsPixelPlugin.getVisitSitePromptDisplayedNewTabMetric()?.fire()
+    }
+
+    override suspend fun fireFirstSiteSuggestionPixel() {
+        onboardingExperimentMetricsPixelPlugin.getFirstSiteSuggestionMetric()?.fire()
+    }
+
+    override suspend fun fireSecondSiteSuggestionPixel() {
+        onboardingExperimentMetricsPixelPlugin.getSecondSiteSuggestionMetric()?.fire()
+    }
+
+    override suspend fun fireThirdSiteSuggestionPixel() {
+        onboardingExperimentMetricsPixelPlugin.getThirdSiteSuggestionMetric()?.fire()
+    }
+
+    override suspend fun fireTrackersBlockedMessageDisplayedPixel() {
+        onboardingExperimentMetricsPixelPlugin.getTrackersBlockedMessageDisplayedMetric()?.fire()
+    }
+
+    override suspend fun fireNoTrackersMessageDisplayedPixel() {
+        onboardingExperimentMetricsPixelPlugin.getNoTrackersMessageDisplayedMetric()?.fire()
+    }
+
+    override suspend fun fireTrackerNetworkMessageDisplayedPixel() {
+        onboardingExperimentMetricsPixelPlugin.getTrackerNetworkMessageDisplayedMetric()?.fire()
+    }
+
+    override suspend fun firePrivacyDashClickedFromOnboardingPixel() {
+        onboardingExperimentMetricsPixelPlugin.getPrivacyDashClickedFromOnboardingMetric()?.fire()
+    }
+
+    override suspend fun fireFireButtonPromptDisplayedPixel() {
+        onboardingExperimentMetricsPixelPlugin.getFireButtonPromptDisplayedMetric()?.fire()
+    }
+
+    override suspend fun fireFireButtonClickedFromOnboardingPixel() {
+        onboardingExperimentMetricsPixelPlugin.getFireButtonClickedFromOnboardingMetric()?.fire()
+    }
+
+    override suspend fun fireFinalOnboardingScreenDisplayedPixel() {
+        onboardingExperimentMetricsPixelPlugin.getFinalOnboardingScreenDisplayedMetric()?.fire()
+    }
+
+    override suspend fun fireSecondSiteVisitPixel() {
+        onboardingDesignExperimentCountDataStore.getSiteVisitCount().takeIf { it < 2 }?.let {
+            onboardingDesignExperimentCountDataStore.increaseSiteVisitCount().takeIf { it == 2 }?.apply {
+                onboardingExperimentMetricsPixelPlugin.getSecondSiteVisitMetric()?.fire()
+            }
+        }
+    }
+
+    override suspend fun fireSecondSerpVisitPixel() {
+        onboardingDesignExperimentCountDataStore.getSerpVisitCount().takeIf { it < 2 }?.let {
+            onboardingDesignExperimentCountDataStore.increaseSerpVisitCount().takeIf { it == 2 }?.apply {
+                onboardingExperimentMetricsPixelPlugin.getSecondSearchVisitMetric()?.fire()
+            }
+        }
+    }
+
+    private suspend fun checkAndSetCachedCohort() {
+        withContext(dispatcherProvider.io()) {
+            cachedOnboardingDesignExperimentCohort = getEnrolledAndEnabledExperimentCohort()
+        }
+    }
+
+    private suspend fun getEnrolledAndEnabledExperimentCohort(): OnboardingDesignExperimentCohort? {
```

Comment:
This is not really doing what it says, you are only getting the cohort but the experiment is not necessarily enabled, you'd have to check for isEnabled() too.